packages/cozy-sharing/src/components/ShareRestrictionModal/ShareRestrictionModal.jsx

```diff
@@ -73,41 +79,49 @@ export const ShareRestrictionModal = ({ file, onClose }) => {
     setLoading(true)
     // If the file is not shared, we create a new sharing link
     if (!hasSharingLink) {
-      const verbs = editingRights === 'readOnly' ? READ_ONLY_PERMS : WRITE_PERMS
-      await shareByLink(file, { verbs })
-      const url = getSharingLink(file._id)
-      await copyToClipboard(url, { t, showAlert })
-      onClose()
-      return
-    }
-
-    await updatePermissions({
-      file,
-      t,
-      editingRights,
-      documentType,
-      updateDocumentPermissions,
-      showAlert
-    })
-
-    const ttl = makeTTL(dateToggle && selectedDate)
-    if (isDesktopOrMobileWithoutShareAPI) {
-      const url = await makeSharingLink(client, [file._id], {
+      const ttl = makeTTL(dateToggle && selectedDate)
+      const { data: perms } = await createPermissions({
+        file,
+        t,
         ttl,
-        password
+        password,
+        editingRights,
+        documentType,
+        shareByLink,
+        showAlert
+      })
```

Comment:
nit: Maybe only `createPermissions` and `updatePermissions` should be in the condition, because the rest is identical between the two conditions ?